### PR TITLE
Login and logout routes

### PR DESCRIPTION
**Pull request by Ale573**
Initial code for login and logout.  Code is in app.py, user dao, and user handler.

### CODE REVIEW

**Review by yetsielaviles — APPROVED** [review]

**Review by jaortiz117 — APPROVED** [review]
Looks good! the way you did it (with flask session) is what I would have done. Just make sure that the json sent from the front end matches the handler

File: handler/user.py [hunk 1]
```diff
@@ -21,6 +22,32 @@ def verify_params(params, required_params):
 
 class UserHandler(BaseHandler):
 
+    @staticmethod
+    def do_login(json):
+        try:
+            user = User.do_login(json)
+            password = json['password']
+            if user and bcrypt.checkpw(password.encode('utf-8'), user.password.encode('utf-8')):
```
Comment (Everod52) [inline handler/user.py:30]: Isn't the `user.password` already encoded? Or is that how the method `checkpw` works?

Comment (Ale573) [reply handler/user.py:30]: That is how the method works. If I removed the encode part it doesn't work.  There is a Server error.

Comment (Everod52) [reply handler/user.py:30]: Alright, well everything looks good. Nice job!

**Review by Everod52 — APPROVED** [review]

**Pull request merged**